include/oneapi/dpl/pstl/tuple_impl.h

```diff
@@ -424,16 +424,36 @@ struct tuple<T1, T...>
         return *this;
     }
 
-    friend bool
-    operator==(const tuple& __lhs, const tuple& __rhs)
-    {
-        return __lhs.holder.value == __rhs.holder.value && __lhs.next == __rhs.next;
-    }
-    friend bool
-    operator!=(const tuple& __lhs, const tuple& __rhs)
-    {
-        return !(__lhs == __rhs);
-    }
+// This macro is used to define binary operators for comparison between oneAPI::dpl::__internal::tuple and
+//  std::tuple when they have the same number of template arguments and the individual elements which are invdividually
+//  comparable. These overloads are required because for overload resolution of template arguments, exact match is
+//  required and the compiler does not consider implicit conversions. We also allow std::tuple implementation to
+//  implement the operators themselves, which guarantees a match with oneAPI::dpl::__internal::tuple
+#define _TUPLE_BINARY_OPERATOR_OVERLOAD(__OPERATOR)                                                                    \
+    template <typename _U1, typename... _U, typename = ::std::enable_if_t<(sizeof...(_U) == sizeof...(T))>>            \
+    friend bool operator __OPERATOR(const tuple& __lhs, const oneapi::dpl::__internal::tuple<_U1, _U...>& __rhs)       \
+    {                                                                                                                  \
+        return static_cast<std::tuple<T1, T...>>(__lhs) __OPERATOR static_cast<std::tuple<_U1, _U...>>(__rhs);         \
```

Comment:
Slightly pedantic, but `std::` should be `::std::` to match the rest of the methods in this class.

Reply:
So, I think our mindset has changed on `::std` vs `std`.  Tagging @MikeDvorskiy.  We had talked about adding this to some sort of a style document or at least making an announcement, but that hasn't happened yet.
 
I was told that it is unnecessary to use `::std` vs `std` and that new code should be the later.  Similar to formatting though, we don't want to ruin the "blame" by changing it aggressively everywhere.

Reply:
Yes, we need to propagate that decision out and also codify it somewhere here in the repo in some sort of a style guide, I think.  We could even simply add a section to `CONTRIBUTING.md` with some general rules and best practices.